cmake/detray-compiler-options.cmake

```diff
@@ -0,0 +1,44 @@
+# Detray library, part of the ACTS project (R&D line)
+#
+# (c) 2021 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
+
+# Include the helper function(s).
+include( detray-functions )
+
+# Set the language standards to use.
+set( CMAKE_CXX_STANDARD 17 CACHE STRING "The (Host) C++ standard to use" )
```

Comment:
In general, building some C\+\+ code with one standard, and others with something else, would not be my first choice.

This current setting, using the cache variable approach, allows us to try out new standards "globally", with the following formalism:

```
cmake -DCMAKE_CXX_STANDARD=20 -S detray -B build ...
```

I think it should be enough if we switch between standards globally, like this. But if absolutely necessary, a given subdirectory could still do something silly like:

```cmake
set( DETRAY_DISPLAY_CXX_STANDARD 20 CACHE STRING "C++ standard for the detray::display library" )
set( CMAKE_CXX_STANDARD ${DETRAY_DISPLAY_CXX_STANDARD} )
```

Since "regular" variables can override cache variables in a "scoped way" with modern CMake. But this would not be my first, or even my second choice... :stuck_out_tongue:

Reply:
Don't worry, I was not proposing that we actually do this :). Mainly I want to play with the ranges library to satisfy my own curiosity, so on a "crazy branch" or something